Common/Util/LeanData.cs

```diff
@@ -238,8 +239,25 @@ public static string GenerateLine(IBaseData data, SecurityType securityType, Res
                             throw new ArgumentOutOfRangeException("resolution", resolution, null);
                     }
                     break;
-            }
 
+                case SecurityType.Base:
+
+                    var vol = data as ForexVolume;
```

Comment:
As the ForexVolume is a custom data type quite specific to FXCM this would be classified as a custom data type. To avoid polluting the LEAN code base we'd prefer these are kept isolated. Imagine we applied the code pattern below to the 1 Million custom data types in quandl -- it would be a mess! So best to do any CSV writing inside the project and not make any changes to *LEAN* Data writer. If required you can tweak the ToCSV to be public and access it from your ForexVolume Toolbox project.